lib/improver/blending/spatial_weights.py

```diff
@@ -0,0 +1,275 @@
+        """
+        Multiply two cubes together by taking slices along the coordinate
+        matching the collapsing_coord string.
+
+        Args:
+            masked_weights_cube (iris.cube.Cube):
+                A cube with spatial weights and any other leading dimensions.
+                This cube must have a coordinate matching the name given by
+                collapsing_coord and this coordinate must be the same length
+                as the corresponding coordinate in the
+                one_dimensional_weights_cube.
+            one_dimensional_weights_cube (iris.cube.Cube):
+                A cube with one_dimensional weights. The only dimension
+                coordinate in this cube matches the string given by
+                collapsing_coord and the length of this coord must match the
+                length of the same coordinate in masked_weights_cube.
+            collapsing_coord (string):
+                The string that will match to a coordinate in both input cube.
+                This is the coordinate that the input cubes will be sliced
+                along and then multipled. The corresponds to the coordinate
+                used to collapse a cube using the weights generated by this
+                plugin.
+
+        Returns:
+            result (iris.cube.Cube):
+                A cube with the same dimensions as the input cube, but with
+                the weights multiplied by the weights from the
+                one_dimensional_weights_cube. The collapsing_coord will be the
+                leading dimension on the output cube.
+        """
+        result = iris.cube.CubeList()
+        if (masked_weights_cube.coord(collapsing_coord) !=
+                one_dimensional_weights_cube.coord(collapsing_coord)):
+            message = ("The collapsing_coord {} does not match on "
+                       "masked_weights_cube and "
+                       "one_dimensional_weights_cube".format(collapsing_coord))
+            raise ValueError(message)
+        for masked_weight_slice, one_dimensional_weight in zip(
+                masked_weights_cube.slices_over(collapsing_coord),
+                one_dimensional_weights_cube.slices_over(collapsing_coord)):
+            masked_weight_slice.data = (
+                masked_weight_slice.data * one_dimensional_weight.data)
+            result.append(masked_weight_slice)
+        result = result.merge_cube()
+        return result
+
+    @staticmethod
+    def normalised_masked_weights(weights_cube, collapsing_coord):
+        """
+        Normalise spatial weights along dimension associated with the
+        collapsing_coord. If for a given point the sum of the weights along
+        the collapsing_coord is zero then the returned normalised weight for
+        that point will also be zero. This correspsonds to the case where
+        there is missing data for that point for all slices along the
+        collapsing_coord.
```

Comment:
I thought we already had a function to do this, somewhere in (`blending.weights.WeightsUtilities`?  Would it be possible to add the spatially varying functionality to the existing utility rather than creating a whole new one?  Or is that more complicated than it seems?